test/exe/global_deploy_test.rb

```diff
@@ -23,13 +23,38 @@ def test_deploy_passes_verify_result
     krane_global_deploy!(flags: '--verify-result false')
   end
 
+  def test_deploy_parses_std_in_without_filenames
+    Dir.mktmpdir do |tmp_path|
+      $stdin.expects("read").returns("")
+      Dir.expects(:mktmpdir).with("krane").yields(tmp_path)
+      set_krane_global_deploy_expectations!(new_args: { filenames: [tmp_path] })
+      krane_global_deploy!(flags: '--stdin')
+    end
+  end
```

Comment:
We need an integration test that actually reads something from STDIN (not just `""`) and check to make sure we output what we expect.

Reply:
Actually, we don't need an integration test. This, along with optionshelper tests are enough to prove this works as intended